src/ReachSets/ContinuousPost/BFFPS19/reach_blocks.jl

```diff
@@ -0,0 +1,301 @@
+    X_store = combine_cpas(X_store, X_store_d, blocks, diff_blocks)
+    store!(res, 1, X_store, t0, t1, N)
+    terminate, skip = termination(1, X_store, t0)
+    if terminate
+        return 1, skip
+    end
+
+    b = length(blocks)
+    Xhatk = Vector{LazySet{NUM}}(undef, b)
+    Xhatk_d = Vector{LazySet{NUM}}(undef, bd)
+    ϕpowerk = copy(ϕ)
+
+    if U != nothing
+        Whatk_blocks = Vector{LazySet{NUM}}(undef, b)
+        inputs = next_set(U)
+        @inbounds for i in 1:b
+            bi = partition[blocks[i]]
+            Whatk_blocks[i] = overapproximate_inputs(1, blocks[i], proj(bi, n) * inputs)
+       end
+    end
+
+    if U != nothing
+       Whatk_diff_blocks = Vector{LazySet{NUM}}(undef, bd)
+       inputs = next_set(U)
+       @inbounds for i in 1:bd
+           bi = partition[diff_blocks[i]]
+           Whatk_diff_blocks[i] = overapproximate_inputs(1, diff_blocks[i], proj(bi, n) * inputs)
+        end
+    end
+
+    k = 2
+    @inbounds while true
+        update!(progress_meter, k)
+        for i in 1:b
+            bi = partition[blocks[i]]
+            Xhatk_bi = ZeroSet(length(bi))
+            for (j, bj) in enumerate(partition)
+                block = ϕpowerk[bi, bj]
+                if !iszero(block)
+                    Xhatk_bi = Xhatk_bi + block * Xhat0[j]
+                end
+            end
+            Xhatk_bi_lazy = (U == nothing ? Xhatk_bi : Xhatk_bi + Whatk_blocks[i])
+            Xhatk[i] = (output_function == nothing) ?
+                overapproximate(blocks[i], Xhatk_bi_lazy) :
+                Xhatk_bi_lazy
+        end
+        array = CartesianProductArray(copy(Xhatk))
+        X_store = (output_function == nothing) ?
+            array :
+            box_approximation(output_function(array))
+
+
+        terminate, skip, rs = termination(k, X_store, t0)
+
+        if  !(isdisjoint(X_store, UnionSetArray(guards_proj)))
+            for i in 1:bd
+                bi = partition[diff_blocks[i]]
+                Xhatk_bi = ZeroSet(length(bi))
+                for (j, bj) in enumerate(partition)
+                    block = ϕpowerk[bi, bj]
+                    if !iszero(block)
+                        Xhatk_bi = Xhatk_bi + block * Xhat0[j]
+                    end
+                end
+                Xhatk_bi_lazy = (U == nothing ? Xhatk_bi : Xhatk_bi + Whatk_diff_blocks[i])
+                Xhatk[i] = (output_function == nothing) ?
+                    overapproximate(diff_blocks[i], Xhatk_bi_lazy) :
+                    Xhatk_bi_lazy
+            end
+
+             array_d = CartesianProductArray(copy(Xhatk_d))
+
+             X_store_d = (output_function == nothing) ?
+                            array_d :
+                            box_approximation(output_function(array_d))
+            if !terminate
+                X_store = combine_cpas(LazySets.Approximations.overapproximate(rs, CartesianProductArray, block_options), X_store_d, blocks, diff_blocks)
```

Comment:
Yes, I removed intersection with invariant from discrete post. Internally it takes an intersection using Polyhedra library anyway, so outsoursing it to the discrete post doesn't really help.

Reply:
It is more of a consistency thing. You need to know about this exception, which makes understanding harder. But I think here it makes sense because you have the low-dimensional sets computed already. I just found that you need to do the same thing the first time you call `termination` (see other comment). Better outsource the common code to a function then.